### DISCUSSION TITLE
Create Admin UI using withKeystone package

### DISCUSSION THREAD

**Discussion by junaid33** [keystonejs/keystone#7893] [Questions]
The app I'm building is using the next-embedded example which creates the query API and graphQL API for Next.js deployment. Would it be possible for the withKeystone package to create the Admin UI as well?

On a side note: Positioning Keystone as a Next.js "plugin" will really drive adoption. By adding a keystone.js file and adding withKeystone in the next config, Next.js users can get a GraphQL API and Admin UI out of the box. That's a really good value add and Next.js users can start using Keystone immediately.

**Comment by flexdinesh** [comment]
Embedded Keystone example limits the Keytone Admin UI to local environments.

There are few technical limitations in getting Keystone to work as a Next.js plugin. Next.js APIs when hosted in Vercel (usually) work in a serverless infra. Keystone currently is not serverless compatible and that's why to use all that Keystone has to offer, Keystone needs to be hosted in a running node server and that's why Keystone can't be built as a plugin for Next.js yet. But we are hopeful that this might change sometime in the near future.

**Reply by junaid33** [reply]
Yes, but Vercel isn’t the only way to deploy Next.js. Next.js on Render, Railway, Fly.io, etc. is deploying to a long running node server.
